configure.ac

```diff
@@ -2387,8 +2383,16 @@ AS_IF([test "x$enable_zph_qos" = "xyes"],[
     [Enable support for QOS netfilter mark preservation])
 ])
 
-AC_CHECK_LIB(regex, regexec, [REGEXLIB="-lregex"],[REGEXLIB=''])
-AC_SUBST(REGEXLIB)
+dnl Look for libregex with regcomp() API
```

Comment:
I do not see any regcomp() checks in added code; have I overlooked them?

AFAICT, removed code was checking for regexec(). If feasible, let's preserve that logic (in this PR). Otherwise, please adjust PR description to explain why we are changing what API we check.